src/server/plugins/blankie.ts

```diff
@@ -1,16 +1,51 @@
 import { type ServerRegisterPluginObject } from '@hapi/hapi'
 import Blankie from 'blankie'
+import flatten from 'lodash/flatten.js'
 
-export default {
-  plugin: Blankie,
-  options: {
-    defaultSrc: ['self'],
-    fontSrc: ['self', 'data:'],
-    connectSrc: ['self'],
-    scriptSrc: ['self', 'unsafe-inline'],
-    styleSrc: ['self', 'unsafe-inline'],
-    imgSrc: ['self'],
-    frameSrc: ['self', 'data:'],
-    generateNonces: false
+import { config } from '~/src/config/index.js'
+
+function combine(...values: string[][]) {
+  return [...new Set(flatten(values))]
+}
+
+const googleAnalyticsOptions = {
+  scriptSrc: ['https://*.googletagmanager.com'],
+  imgSrc: ['https://*.google-analytics.com', 'https://*.googletagmanager.com'],
+  connectSrc: [
+    'https://*.google-analytics.com',
+    'https://*.analytics.google.com',
+    'https://*.googletagmanager.com'
+  ]
+}
+
+export const configureBlankiePlugin = (): ServerRegisterPluginObject<
+  Record<string, boolean | string | string[]>
+> => {
+  const gaTrackingId = config.get('googleAnalyticsTrackingId')
+
+  /*
+  Note that unsafe-inline is a fallback for old browsers that don't support nonces. It will be ignored by modern browsers as the nonce is provided.
+  */
+  return {
+    plugin: Blankie,
+    options: {
+      defaultSrc: ['self'],
+      fontSrc: ['self', 'data:'],
+      connectSrc: combine(
+        ['self'],
+        gaTrackingId ? googleAnalyticsOptions.connectSrc : []
+      ),
```

Comment:
Example using `.flat()`

Won't need Lodash or `new Set()` for this

```suggestion
      connectSrc: [
        ['self'],
        gaTrackingId ? googleAnalyticsOptions.connectSrc : []
      ].flat(),
```

Reply:
I used the set to handle duplicates, in the event of a clash between GA and another dependency. After some refactoring there aren't any duplicates anymore so I'm happy to remove it.